lib/components/Switcher.vue

```diff
@@ -0,0 +1,101 @@
+<template>
+    <label class="switch">
+        <input
+          ref="switcher"
+          type="checkbox"
+          :checked="value"
+          @change="onCheckedChange($event)"
+          @click="updateValue()"
+        />
+        <span class="slider" />
+    </label>
+</template>
+
+<script>
+export default {
+    name: 'Switcher',
```

Comment:
Again, see following link for a very similar example. Need model event set to checkedChange instead